src/ControlSystem/ExpirationTimes.hpp

```diff
@@ -0,0 +1,122 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <algorithm>
+#include <memory>
+#include <string>
+#include <type_traits>
+#include <unordered_map>
+
+#include "ControlSystem/Tags.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "Domain/Creators/DomainCreator.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace control_system {
+/*!
+ * \ingroup ControlSystemGroup
+ * \brief Construct the initial expiration times for functions of time that are
+ * controlled by a control system
+ *
+ * The expiration times are constructed using inputs from control system
+ * OptionHolders as an unordered map from the name of the function of time being
+ * controlled to the expiration time. The expiration time is computed as
+ * \f$\tau_\mathrm{exp} = \alpha_\mathrm{update} \tau_\mathrm{damp}\f$ where
+ * \f$\alpha_\mathrm{update}\f$ is the update fraction supplied as input to the
+ * Controller and \f$\tau_\mathrm{damp}\f$ is/are the damping timescales
+ * supplied from the TimescaleTuner (\f$\tau_\mathrm{damp}\f$ is a DataVector
+ * with as many components as the corresponding function of time, thus
+ * \f$\tau_\mathrm{exp}\f$ will also be a DataVector of the same length).
+ *
+ * To protect against bad inputs, if the initial expiration time that is
+ * calculated is smaller than the initial time step, then the expiration time is
+ * simply set to the initial time step. However, the MeasurementTimescales have
```

Comment:
I've been using this for all the runs I've been doing since this was originally merged over a year ago, and it hasn't been an issue so far so I'll leave it for now. If it becomes an issue, we can deal with it then.